### PR TITLE
[OneCollectorExporter] Initial code

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

Initial check-in of OneCollectorExporter. I'm expecting this to change a lot but I want to get a preview version out to some customers/early adopters so they can start playing with it.

### CODE REVIEW

File: .github/component_owners.yml [hunk 1]
```diff
@@ -17,6 +17,13 @@ components:
     - Yun-Ting
   src/OpenTelemetry.Exporter.Instana/:
     - zivaninstana
+  src/OpenTelemetry.Exporter.OneCollector/:
+    - cijothomas
```
Comment (reyang) [inline .github/component_owners.yml:21]: Scope this down to just two of us?

File: src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporterTransportOptions.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    /// cref="OneCollectorExporterTransportProtocolType.HttpJsonPost"/>.
+    /// </summary>
+    internal OneCollectorExporterTransportProtocolType Protocol { get; set; } = OneCollectorExporterTransportProtocolType.HttpJsonPost;
+
+    /// <summary>
+    /// Gets or sets the maximum request payload size in bytes when sending data
+    /// to OneCollector. Default value: 4,194,304.
+    /// </summary>
+    /// <remarks>
+    /// Note: Set to -1 for unlimited request payload size.
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporterTransportOptions.cs:53]: Do we want unlimited?

File: src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporter.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Exporter.OneCollector;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Exporter;
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporter.cs:21]: Which one do we want `OpenTelemetry.Exporter` or `OpenTelemetry.Exporter.OneCollector`?

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporter.cs (hunk 3 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporter.cs]: Good question. Not totally sure! Kind of inconsistent...

* [Zipkin uses OpenTelemetry.Exporter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/d0829fff2f229b67c884526cb4943b99f6c75f29/src/OpenTelemetry.Exporter.Zipkin/ZipkinExporter.cs#L30) as does Jaeger & Otlp
* [Prometheus uses OpenTelemetry.Exporter.Prometheus but isn't public](https://github.com/open-telemetry/opentelemetry-dotnet/blob/d0829fff2f229b67c884526cb4943b99f6c75f29/src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusExporter.cs#LL20C11-L20C44)
* [Geneva uses OpenTelemetry.Exporter.Geneva](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/9c89abf4780fe17508c8cb9d78d716df9e6663ab/src/OpenTelemetry.Exporter.Geneva/GenevaTraceExporter.cs#LL23C11-L23C40)

I don't recall if any of that was intention (@alanwest?). But "OpenTelemetry.Exporter.[Name]" seems reasonable to me so I updated the PR with one exception:

The current extension (`OneCollectorOpenTelemetryLoggerOptionsExtensions.AddOneCollectorExporter`) I left in `OpenTelemetry.Logs`. [Geneva uses Microsoft.Extensions.Logging](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/9c89abf4780fe17508c8cb9d78d716df9e6663ab/src/OpenTelemetry.Exporter.Geneva/GenevaLoggingExtensions.cs#L23) but that seems kind of weird to me. My practice generally is to put the extensions in the namespace of the thing you are extending.

Comment (reyang) [reply src/OpenTelemetry.Exporter.OneCollector/OneCollectorExporter.cs]: @CodeBlanch the updates you made make a lot of sense to me 👍

File: test/OpenTelemetry.Exporter.OneCollector.Tests/WriteDirectlyToTransportSinkTests.cs [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+            serializationResult = new BatchSerializationResult
+            {
+                NumberOfItemsSerialized = numberOfItemsSerialized,
+                PayloadSizeInBytes = stream.Length,
+            };
+        }
+    }
+    */
+
```
Comment (Kielek) [inline test/OpenTelemetry.Exporter.OneCollector.Tests/WriteDirectlyToTransportSinkTests.cs:227]: Do you rellay need this code?

```suggestion
```

Comment (CodeBlanch) [reply test/OpenTelemetry.Exporter.OneCollector.Tests/WriteDirectlyToTransportSinkTests.cs]: Woops sorry it was used at one point but I refactored it out. Removed 😄

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Comment by julealgon** [comment]
Can somebody in here provide some more details on this "OneCollector" project? Searched for it but can't find anything at all. Maybe add a link to the readme page for the exporter?

**Comment by CodeBlanch** [comment]
@julealgon Opened this PR: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1015

**Pull request merged**